components/builder-api/src/server/resources/events.rs

```diff
@@ -0,0 +1,185 @@
+                    channel: Query<ToChannel>,
+                    date_range: Query<DateRange>,
+                    search_query: Query<SearchQuery>)
+                    -> HttpResponse {
+    match do_get_events(&req, &pagination, &channel, &date_range, &search_query) {
+        Ok((events, count)) => postprocess_event_list(&req, &events, count, &pagination),
+        Err(err) => {
+            error!("{}", err);
+            err.into()
+        }
+    }
+}
+
+#[allow(clippy::needless_pass_by_value)]
+async fn get_events_from_saas(req: HttpRequest,
+                              pagination: Query<Pagination>,
+                              channel: Query<ToChannel>,
+                              date_range: Query<DateRange>,
+                              search_query: Query<SearchQuery>,
+                              state: Data<AppState>)
+                              -> HttpResponse {
+    let bldr_url = &state.config.api.saas_bldr_url;
+
+    let mut headers = reqwest::header::HeaderMap::new();
+    headers.insert(USER_AGENT_BLDR.0.clone(), USER_AGENT_BLDR.1.clone());
+    if req.headers().contains_key(http::header::AUTHORIZATION) {
+        headers.insert(http::header::AUTHORIZATION,
+                       req.headers()
+                          .get(http::header::AUTHORIZATION)
+                          .unwrap()
+                          .clone());
+    }
+
+    let http_client = match HttpClient::new(bldr_url, headers) {
+        Ok(client) => client,
+        Err(err) => {
+            debug!("HttpClient Error: {:?}", err);
+            return HttpResponse::InternalServerError().body(err.to_string());
+        }
+    };
+
+    // We are expecting dates in YYYY-MM-DD format
+    let from_date = date_range.from_date.date().format("%Y-%m-%d").to_string();
+    let to_date = date_range.to_date.date().format("%Y-%m-%d").to_string();
+    let url = format!("{}/v1/depot/events?range={}&channel={}&from_date={}&to_date={}&query={}",
+                      bldr_url,
+                      pagination.range,
+                      channel.channel,
+                      from_date,
+                      to_date,
+                      search_query.query);
+    debug!("SaaS Url: {}", url);
+    match http_client.get(&url)
+                     .send()
+                     .await
+                     .map_err(Error::HttpClient)
+    {
+        Ok(response) => {
+            match response.text().await {
+                Ok(body) => {
+                    let mut http_response = HttpResponse::Ok();
+
+                    http_response.header(http::header::CONTENT_TYPE, headers::APPLICATION_JSON)
+                                 .header(http::header::CACHE_CONTROL, headers::NO_CACHE)
+                                 .body(body)
+                }
+                Err(err) => {
+                    debug!("Error getting response text: {:?}", err);
+                    HttpResponse::InternalServerError().body(err.to_string())
```

Comment:
So sorry. I should have been more clear. I meant something like:

```
let msg = format!("Error getting response text: {:?}", err);
debug!(msg);
HttpResponse::InternalServerError().body(msg)
```

So that the response body gets the detail.